Dockerfile

```diff
@@ -1,38 +1,58 @@
-FROM ubuntu:trusty
-
-RUN apt -y update && \
-  apt -y install \
+FROM ubuntu:18.04
+RUN apt-get update
+RUN apt-get install -y \
+    autoconf \
     autoconf-archive \
-    libcmocka0 \
-    libcmocka-dev \
+    automake \
     build-essential \
-    wget \
+    g++ \
+    gcc \
     git \
     libssl-dev \
-    pkg-config \
-    gcc \
-    g++ \
-    m4 \
     libtool \
-    automake \
-    autoconf
+    m4 \
+    net-tools \
+    pkg-config \
+    wget
 
-RUN wget https://downloads.sourceforge.net/project/ibmswtpm2/ibmtpm532.tar && \
-  sha256sum ibmtpm532.tar | grep -q ^abc0b420257917ccb42a9750588565d5e84a2b4e99a6f9f46c3dad1f9912864f && \
-  mkdir ibmtpm532 && \
-  tar axf ibmtpm532.tar -C ibmtpm532 && \
-  make -C ibmtpm532/src -j$(nproc)
+# OpenSSL
+ARG openssl_name=openssl-1.1.0h
+WORKDIR /tmp
+ADD https://www.openssl.org/source/$openssl_name.tar.gz .
+RUN tar xvf $openssl_name.tar.gz
+WORKDIR $openssl_name
+RUN ./config --prefix=/usr/local/openssl --openssldir=/usr/local/openssl
+RUN make -j$(nproc)
+RUN make install
+RUN openssl version
 
-COPY . tpm2-tss
+# IBM's Software TPM 2.0
+ARG ibmtpm_name=ibmtpm1119
+WORKDIR /tmp
+RUN wget "https://downloads.sourceforge.net/project/ibmswtpm2/$ibmtpm_name.tar.gz"
```

Comment:
For constancy with the above openssl it would be good to make this an ADD as well (also because then docker knows how to download it through daemon configured proxies)
`ADD https://downloads.sourceforge.net/project/ibmswtpm2/$ibmtpm_name.tar.gz .`

Reply:
Thanks for this update!

Reply:
That makes sense. I've made the change you suggested.

I originally had ``RUN wget "https://sourceforge.net/projects/ibmswtpm2/files/$ibmtpm_name.tar.gz/download" -O $ibmtpm_name.tar.gz`` as I didn't know about the alternate Sourceforge URL.

Reply:
Ah yup, I do that to a lot. Thanks!